File: leetcode/49. Group Anagrams/GroupAnagrams.go
func groupAnagrams(strs []string) [][]string {

	res := make([][]string, 0)
	groups := make(map[string][]string)

	for _, str := range strs {
		var charCount [26]int

		for _, chr := range str {
			charCount[chr-'a']++
		}

		key := ""

		for i, v := range charCount {
			pair := strconv.Itoa(i) + "," + strconv.Itoa(v)
			key += pair
		}
		groups[key] = append(groups[key], str)
	}
	for _, group := range groups {
		res = append(res, group)
	}
	return res
}

File: leetcode/49. Group Anagrams/GroupAnagrams.cpp
class Solution
{
public:
    vector<vector<string>> groupAnagrams(vector<string> &strs)
    {

        vector<vector<string>> result;
        unordered_map<string, vector<string>> m;

        for (const string str : strs)
        {
            string key = "00000000000000000000000000";
            for (const char chr : str)
            {
                key[chr - 'a']++;
            }
            m[key].push_back(str);
        }
        for (auto elem : m)
        {
            result.push_back(elem.second);
        }
        return result;
    }
};